core/src/main/kotlin/org/evomaster/core/search/gene/mongo/ObjectIdGene.kt

```diff
@@ -0,0 +1,113 @@
+package org.evomaster.core.search.gene.mongo
+
+import org.evomaster.core.logging.LoggingUtil
+import org.evomaster.core.output.OutputFormat
+import org.evomaster.core.search.gene.*
+import org.evomaster.core.search.gene.regex.*
+import org.evomaster.core.search.gene.root.CompositeFixedGene
+import org.evomaster.core.search.gene.utils.GeneUtils
+import org.evomaster.core.search.service.Randomness
+import org.evomaster.core.search.service.mutator.genemutation.AdditionalGeneMutationInfo
+import org.evomaster.core.search.service.mutator.genemutation.SubsetGeneMutationSelectionStrategy
+import org.slf4j.Logger
+import org.slf4j.LoggerFactory
+
+/**
+ * In MongoDB, documents stored in a collection require a unique _id field that acts as a primary key.
+ * ObjectIds are used as the default value for the _id field
+ */
+class ObjectIdGene(
+    /**
+     * The name of this gene
+     */
+    name: String,
+
+    private val id: RegexGene = RegexGene(
```

Comment:
instead of creating it by hand, should use `RegexHandler.createGeneForJVM(regex)`. this will avoid issues in future, in case we change regex genes. also, it is more readable